src/common/components/text-input/TextInput.tsx

```diff
@@ -15,33 +15,63 @@ interface Props {
   label?: string;
   placeholder?: string;
   onChange?: ChangeEventHandler<HTMLInputElement>;
-  rules?: { pattern: RegExp };
+  rules?: {
+    pattern: {
+      value: RegExp;
+      message: string;
+    };
+  };
+  setError: any;
+  setValue: any;
+  showValueInLabel: boolean;
 }
 
 const _TextInput: ForwardRefRenderFunction<HTMLInputElement, Props> = (
   props,
   externalRef
 ) => {
   const {
-    field,
-    // fieldState
-  } = useController(props);
+    field: { ref, ...inputProps },
+    fieldState,
+  } = useController({
+    name: props.name,
+    control: props.control,
+    rules: props.rules,
+    defaultValue: "",
+  });
 
   return (
     <StyledInput className="TextInput">
       <label className="label">{props.label}</label>
       <div>
         <input
-          {...field}
+          {...inputProps}
           placeholder={props.placeholder}
-          ref={field.ref}
-          value={field.value}
-          onChange={props.onChange ?? field.onChange}
+          onChange={(e) => {
+            const value = e.target.value;
+            props.setValue(props.name, value);
+            if (props.rules) {
+              const regExp = new RegExp(props.rules?.pattern.value);
+              regExp.test(value);
+              if (!regExp.test(value)) {
+                props.setError(props.name, {
+                  message: props.rules.pattern.message,
```

Comment:
is the intention here to show an error like under the box or something but allow any input?